File: app/src/main/java/org/tvheadend/tvhclient/features/streaming/external/CastRemoteMediaClientCallback.java
package org.tvheadend.tvhclient.features.streaming.external;

import com.google.android.gms.cast.framework.media.RemoteMediaClient;

import timber.log.Timber;

class CastRemoteMediaClientCallback extends RemoteMediaClient.Callback {

    private final RemoteMediaClient remoteMediaClient;

    CastRemoteMediaClientCallback(RemoteMediaClient remoteMediaClient) {
        this.remoteMediaClient = remoteMediaClient;
    }

    @Override
    public void onStatusUpdated() {
        Timber.d("onStatusUpdated");
        remoteMediaClient.unregisterCallback(this);
    }

    @Override
    public void onMetadataUpdated() {
        Timber.d("onMetadataUpdated");
    }

    @Override
    public void onQueueStatusUpdated() {
        Timber.d("onQueueStatusUpdated");
    }

    @Override
    public void onPreloadStatusUpdated() {
        Timber.d("onPreloadStatusUpdated");
    }

    @Override
    public void onSendingRemoteMediaRequest() {
        Timber.d("onSendingRemoteMediaRequest");
    }

    @Override
    public void onAdBreakStatusUpdated() {
        Timber.d("onAdBreakStatusUpdated");
    }
}

File: app/src/main/java/org/tvheadend/tvhclient/ui/features/playback/external/CastRemoteMediaClientCallback.kt
package org.tvheadend.tvhclient.ui.features.playback.external

import com.google.android.gms.cast.framework.media.RemoteMediaClient

import timber.log.Timber

internal class CastRemoteMediaClientCallback(private val remoteMediaClient: RemoteMediaClient) : RemoteMediaClient.Callback() {

    override fun onStatusUpdated() {
        Timber.d("onStatusUpdated")
        remoteMediaClient.unregisterCallback(this)
    }

    override fun onMetadataUpdated() {
        Timber.d("onMetadataUpdated")
    }

    override fun onQueueStatusUpdated() {
        Timber.d("onQueueStatusUpdated")
    }

    override fun onPreloadStatusUpdated() {
        Timber.d("onPreloadStatusUpdated")
    }

    override fun onSendingRemoteMediaRequest() {
        Timber.d("onSendingRemoteMediaRequest")
    }

    override fun onAdBreakStatusUpdated() {
        Timber.d("onAdBreakStatusUpdated")
    }
}
